DoLahTest/CommandUnitTest.cpp

```diff
@@ -89,6 +91,33 @@ namespace DoLahTest
             Assert::AreEqual((size_t) 5, resultVector.size());
         }
 
+        TEST_METHOD(SearchDateTaskCommand) {
+            //Arrange
+            DoLah::Calendar testCal;
+            DoLah::AbstractCommand* command;
+            command = DoLah::CommandParser::parse("add task1 by next week");
+            command->setCalendar(&testCal);
+            command->execute();
+            command = DoLah::CommandParser::parse("add task2 in 2 weeks");
+            command->setCalendar(&testCal);
+            command->execute();
+            command = DoLah::CommandParser::parse("add task3 from today to next week");
+            command->setCalendar(&testCal);
+            command->execute();
+            command = DoLah::CommandParser::parse("done 3");
+            command->setCalendar(&testCal);
+            command->execute();
```

Comment:
This is part of the arrange. (and can made to look simple during the test by abstraction inside a builder or factory).